plugin-api/src/main/java/org/hyperledger/besu/plugin/services/storage/KeyValueStorage.java

```diff
@@ -70,6 +71,15 @@
    */
   long removeAllKeysUnless(Predicate<byte[]> retainCondition) throws StorageException;
 
+  /**
+   * Performs an evaluation against each key in the store, returning the set of entries that pass.
+   *
+   * @param returnCondition predicate to evaluate each key against, unless the result is {@code
+   *     null}, the key is added to the returned list of keys.
+   * @return the set of keys that pass the condition.
+   */
+  Set<byte[]> getAllKeysThat(Predicate<byte[]> returnCondition);
```

Comment:
Can't see where this is being used other than in tests. Is it needed?

Reply:
I see now that  it's going used in another PR https://github.com/hyperledger/besu/pull/25

Reply:
this PR is to prepare the rest. The plan was to make it easier for review just to split the storage changes from the rest.